java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeSECheck.java

```diff
@@ -191,7 +195,26 @@ private static ProgramState handleDivisionRemainder(ProgramState previousState,
         // Reuse zero
         return currentState.unstackValue(1).state.stackValue(op1);
       }
-      return currentState.removeConstraintsOnDomain(result, BooleanConstraint.class).addConstraint(result, ZeroConstraint.NON_ZERO);
+      if (isNonZero(currentState, op1)) {
+        return currentState.removeConstraintsOnDomain(result, BooleanConstraint.class).addConstraint(result, ZeroConstraint.NON_ZERO);
+      }
+      return currentState.removeConstraintsOnDomain(result, BooleanConstraint.class);
+    }
+
+    private static ProgramState handleRemainder(ProgramState previousState, ProgramState currentState) {
+      List<SymbolicValue> operands = previousState.peekValues(2);
+      SymbolicValue result = currentState.peekValue();
+      SymbolicValue op1 = operands.get(0);
+      SymbolicValue op2 = operands.get(1);
+      if (isZero(currentState, op2)) {
+        // Division by zero
+        // TODO exceptional yield ?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)